rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,91 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
 
 ## Scope
 
-To be defined.
+### Endpoints
+
+*   An endpoint is defined as a handler interface which receives a request and returns a response or an error to the client.
```

Comment:
From the article linked above, it seems that the benefits of full-duplex communication and multiple channels might make up for the extra configuration burden. Also, I suspect that some sort of authentication is ideal, assuming that different components of the distributed application could be on the public Internet or on shared hosting. And yes, I was assuming longer-lived connections.

There are ways of reducing the public/private key sharing burden. One that comes to mind is using a single key per application. All components of the system can connect (in either direction) to any other component securely. Given that, at some point, some configuration file and/or embedded info is going to be required to discover and connect to some kind of coordinator service, this seems to be a reasonable approach.